aggregate/aggregate

```diff
@@ -0,0 +1,223 @@
+
+    
+    if not string_df.empty and not mentha_df.empty:
+
+        mentha_target_id = mentha_df["target uniprot id"].iloc[0]
+        string_target_id = string_df["target uniprot id"].iloc[0]
+        if mentha_target_id != string_target_id:
+            print(f"Error: Target Uniprot IDs do not match between mentha and string outputs.")
+            exit(1)
+
+        ## MERGING ##
+        # Merge mentha_df and string_df:
+        merged_df = pd.merge(
+            mentha_df, 
+            string_df[["interactor uniprot id", "PDB_ID", "string score"]], 
+            on="interactor uniprot id", 
+            how="left"
+        )
+
+        # Update structure column:
+        merged_df["structure"] = merged_df.apply(
+            lambda row: ",".join(filter(None, sorted(set(row["structure"].split(",") + row["PDB_ID"].split(",")))))
+            if pd.notna(row["PDB_ID"]) else row["structure"],
+            axis=1
+        )
+        
+        # Drop unnecessary columns:
+        merged_df.drop(columns=[ "PDB_ID"], inplace=True)
+
+        merged_df.sort_values(by=["mentha score", "string score"], ascending=[False, False], inplace=True)
+
+        # Find string entries not in mentha:
+        unmatched_string_df = string_df[~string_df["interactor uniprot id"].isin(mentha_df["interactor uniprot id"])]
+        unmatched_string_df = unmatched_string_df.rename(columns={"PDB_ID": "structure"})
+        unmatched_string_df["mentha score"] = None
+        unmatched_string_df = unmatched_string_df[[
+            "target uniprot id", "target uniprot gene", "interactor uniprot id", 
+            "interactor uniprot gene", "mentha score", "string score", "structure"
+        ]]
+
+        # Concatenate the unmatched string entries to df:
+        final_df = pd.concat([merged_df, unmatched_string_df], ignore_index=True)
+
+    elif mentha_df.empty and not string_df.empty:
+        print(f"Warning: No results found in Mentha for the query protein")
+        final_df = string_df
+        # Add mentha score for consistency:
+        final_df["mentha score"] = ""
+
+    elif string_df.empty and not mentha_df.empty:
+        print(f"Warning: No results found in STRING for the query protein")
+        final_df = mentha_df
+        # Add string score for consistency:
+        final_df["string score"] = ""
+
+    else:
+        print(f"Warning:No results found in Mentha and STRING for the query protein")
+        exit(0)
+
+    # Add other resources column (where we will report pdbminer/pdbminer_complexes):
+    final_df["other resources"] = ""
+
+    # Ensure desired column order:
+    final_columns = [
+        "target uniprot id", 
+        "target uniprot gene", 
+        "interactor uniprot id", 
+        "interactor uniprot gene", 
+        "mentha score", 
+        "string score", 
+        "structure", 
+        "other resources"
+    ]
+
+    final_df = final_df.reindex(columns=final_columns)
```

Comment:
What is this supposed to achieve? I'm not quite sure

Reply:
I wanted to re-order the columns, so that mentha score and string score are next ot each other

Reply:
ok, then you can do it like:

```
final_df = final_df[final_columns]
```

which is a bit more explicit/straightforward